src/lib/removal.sh

```diff
@@ -0,0 +1,75 @@
+#! /bin/sh
+
+import logging
+import assert
+
+_rm_name_='removal.sh'
+
+_delete_file() {
+    if [ -e "$1" ]
+    then
+        _debug "Removing $1.."
+        rm "$1"
+        rc=$? # saving rc for the error message
+        if [ $rc -ne 0 ]; then
+            _error "'rm $1' failed! rc=$rc"
+            return 1
+        fi
+    else
+        _warn "Trying to remove $1 but it does not exist! Skipping."
+    fi
+}
+
+delete_files() {
+    failed=0
+    if ! [ -z "$1" ]; then # skip printing the message if empty
+        _info "$1"
+    fi
+    shift  # first argument must always be info string to print, the rest of the arguments are files
+    for var in "$@"
+    do
+        _delete_file "$var"
+        failed=$(($failed + $?))
```

Comment:
You probably don't need the dollar sign within the double parens, like
```suggestion
        failed=$((failed + $?))
```